packages/backend-defaults/src/entrypoints/auditor/Auditor.ts

```diff
@@ -0,0 +1,310 @@
+          Object.keys(options.meta ?? {}).length === 0 &&
+          Object.keys(params?.meta ?? {}).length === 0
+            ? undefined
+            : { ...options.meta, ...params?.meta };
+
+        await this.log({
+          ...options,
+          meta,
+          status: 'succeeded',
+        });
+      },
+      fail: async params => {
+        // return undefined if both objects are empty; otherwise, merge the objects
+        const meta =
+          Object.keys(options.meta ?? {}).length === 0 &&
+          Object.keys(params.meta ?? {}).length === 0
+            ? undefined
+            : { ...options.meta, ...params.meta };
+
+        await this.log({
+          ...options,
+          ...params,
+          meta,
+          status: 'failed',
+        });
+      },
+    };
+  }
+
+  private async getActorId(
+    request?: Request<any, any, any, any, any>,
+  ): Promise<string | undefined> {
+    let credentials: BackstageCredentials =
+      await this.auth.getOwnServiceCredentials();
+
+    if (request) {
+      try {
+        credentials = await this.httpAuth.credentials(request);
+      } catch (error) {
+        throw new ForwardedError('Could not resolve credentials', error);
+      }
+    }
+
+    if (this.auth.isPrincipal(credentials, 'user')) {
+      return credentials.principal.userEntityRef;
+    }
+
+    if (this.auth.isPrincipal(credentials, 'service')) {
+      return credentials.principal.subject;
+    }
+
+    return undefined;
+  }
+
+  private async reshapeAuditorEvent<T extends JsonObject>(
+    options: AuditorEventOptions<T>,
+  ): Promise<AuditorEvent> {
+    const { eventId, severityLevel = 'low', request, ...rest } = options;
+
+    const auditEvent: AuditorEvent = [
+      `${this.plugin.getId()}.${eventId}`,
+      {
+        severityLevel,
+        actor: {
+          actorId: await this.getActorId(request),
+          ip: request?.ip,
+          hostname: request?.hostname,
+          userAgent: request?.get('user-agent'),
+        },
+        request: request
+          ? {
+              url: request?.originalUrl,
+              method: request?.method,
+            }
+          : undefined,
+        ...rest,
+      },
+    ];
+
+    return auditEvent;
+  }
+}
+
+/** @public */
+export class DefaultRootAuditorService {
+  private readonly impl: winston.Logger;
+
+  private constructor(impl: winston.Logger) {
+    this.impl = impl;
+  }
+
+  /**
+   * Creates a {@link DefaultRootAuditorService} instance.
+   */
+  static create(options?: RootAuditorOptions): DefaultRootAuditorService {
+    const defaultFormatter =
+      process.env.NODE_ENV === 'production'
+        ? defaultProdFormat
+        : DefaultRootAuditorService.colorFormat();
+
+    let auditor = winston.createLogger({
+      level: 'info',
+      format: winston.format.combine(
+        auditorFieldFormat,
+        options?.format ?? defaultFormatter,
+      ),
+      transports: options?.transports ?? defaultConsoleTransport,
+    });
```

Comment:
You could use the existing `WinstonLogger` here right? with an explicit `.json()` format.